### PR TITLE
Add test for password reset flow, also get rid of redundant code

### PR DESCRIPTION
**Pull request by CrystalPea**
The new test test password reset flow. A positive side-effect is that it updates `email_access_validated_at` field for our seeded user.

Connected to that: should I also copy this test to staging_and_prod folder, so seeded user `email_access_validated_at` field is also updated in staging and production?

### CODE REVIEW

File: tests/functional/preview_and_dev/test_email_auth.py [hunk 1]
```diff
@@ -10,3 +12,32 @@ def test_email_auth(driver):
     sign_in_email_auth(driver)
     # assert url is research mode service's dashboard
     assert driver.current_url == config['notify_admin_url'] + '/services/{}'.format(config['service']['id'])
+    base_page = BasePage(driver)
+    base_page.sign_out()
+
+
+@recordtime
+def test_reset_password(driver):
```
Comment (idavidmcdonald) [inline tests/functional/preview_and_dev/test_email_auth.py]: Optional: You could rename this `test_reset_forgotten_password` just so it's very clear this is the flow to reset a forgotten password as opposed to just changing your password after you've logged in

File: tests/pages/locators.py [hunk 2]
```diff
@@ -4,6 +4,7 @@
 class CommonPageLocators(object):
     NAME_INPUT = (By.NAME, 'name')
     EMAIL_INPUT = (By.NAME, 'email_address')
+    NEW_PASSWORD_INPUT = (By.NAME, 'new_password')
```
Comment (idavidmcdonald) [inline tests/pages/locators.py]: I'm not sure if I'd put this one in the common page locators section given it feels like it would be pretty rare, I'm assuming only used at most on the change your password form and the reset your forgotten password form

File: tests/test_utils.py [hunk 3]
```diff
@@ -98,14 +98,22 @@ def do_verify(driver):
...
 def do_email_auth_verify(driver):
+    do_email_verification(
+        driver,
+        config['notify_templates']['email_auth_template_id'],
+        config['service']['email_auth_account']
+    )
+
+
+@retry(RetryException, tries=config['verify_code_retry_times'], delay=config['verify_code_retry_interval'])
+def do_email_verification(driver, template_id, email_address):
```
Comment (idavidmcdonald) [inline tests/test_utils.py]: Good refactor here to keep this DRY

File: tests/pages/pages.py [hunk 4]
```diff
@@ -235,11 +236,7 @@ def register(self):
         self.email_input = config['user']['email']
         self.mobile_input = config['user']['mobile']
         self.password_input = config['user']['password']
-        self.click_continue_button()
-
-    def click_continue_button(self):
-        element = self.wait_for_element(RegistrationPage.continue_button)
-        element.click()
+        self.click_continue()
```
Comment (idavidmcdonald) [inline tests/pages/pages.py]: Oh the other thing I forgot to say in my previous review was it would have been nice to see this refactoring of existing functionality done in a separate commit as it makes reviewing easier and makes it clearer what the changes are and why. Don't worry, definitely not the end of the world but always good when you can to do refactoring and adding/editing functionality in separate commits.

File: tests/functional/preview_and_dev/test_email_auth.py (hunk 1 above)
Comment (CrystalPea) [reply tests/functional/preview_and_dev/test_email_auth.py]: Nice one, will do

File: tests/pages/pages.py (hunk 4 above)
Comment (CrystalPea) [reply tests/pages/pages.py]: That's a good advice, I will try to separate my tasks better in the future :)

File: tests/pages/locators.py (hunk 2 above)
Comment (CrystalPea) [reply tests/pages/locators.py]: yeah, you're probably right. And also about password reset not being something that we want happening ever 10 minutes 😅 . We will have to  think of a different strategy for that then.

**Comment by leohemsted** [comment]
we could probs change staging/live functional test user to log in via email rather than sms to avoid the 90 days problem

**Comment by CrystalPea** [comment]
@leohemsted that's a cool idea, I'm all for it :)
